internal/dao/patch.go

```diff
@@ -0,0 +1,47 @@
+package dao
+
+import "encoding/json"
+
+type JsonPatch struct {
+	Spec Spec `json:"spec"`
+}
+
+type Spec struct {
+	Template Template `json:"template"`
+}
+
+type Template struct {
+	Spec ImagesSpec `json:"spec"`
+}
+
+type ImagesSpec struct {
+	SetElementOrders []Element `json:"$setElementOrder/containers"`
+	Containers       []Element `json:"containers"`
+}
+
+type Element struct {
+	Image string `json:"image,omitempty"`
+	Name  string `json:"name"`
+}
+
+// Build a json patch string to update PodSpec images
+func SetImageJsonPatch(images map[string]string) (string, error) {
+	elementOrders := make([]Element, 0)
```

Comment:
Try var elementOrders, containers []Element

Reply:
Also elementOrders seems a bit redundant ie the order is already in the other slice that contains the name and the image?

Reply:
fixed !